src/main/java/uk/gov/companieshouse/disqualifiedofficers/delta/processor/DisqualifiedOfficersDeltaProcessor.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.companieshouse.disqualifiedofficers.delta.processor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Objects;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.delta.DisqualificationDelta;
+import uk.gov.companieshouse.delta.ChsDelta;
+import uk.gov.companieshouse.disqualifiedofficers.delta.exception.RetryableErrorException;
+import uk.gov.companieshouse.disqualifiedofficers.delta.producer.DisqualifiedOfficersDeltaProducer;
+import uk.gov.companieshouse.disqualifiedofficers.delta.transformer.DisqualifiedOfficersApiTransformer;
+
+
+@Component
+public class DisqualifiedOfficersDeltaProcessor {
+
+    private final DisqualifiedOfficersDeltaProducer deltaProducer;
+    private final DisqualifiedOfficersApiTransformer transformer;
+
+    @Autowired
+    public DisqualifiedOfficersDeltaProcessor(DisqualifiedOfficersDeltaProducer deltaProducer,
+            DisqualifiedOfficersApiTransformer transformer) {
+        this.deltaProducer = deltaProducer;
+        this.transformer = transformer;
+    }
+
+    /**
+     * Process CHS Delta message.
+     */
+    public void processDelta(Message<ChsDelta> chsDelta) {
+        try {
+            MessageHeaders headers = chsDelta.getHeaders();
+            final String receivedTopic =
+                    Objects.requireNonNull(headers.get(KafkaHeaders.RECEIVED_TOPIC)).toString();
+            final ChsDelta payload = chsDelta.getPayload();
+
+            ObjectMapper mapper = new ObjectMapper();
+            DisqualificationDelta disqualifiedOfficersDelta = mapper.readValue(payload.getData(),
+                    DisqualificationDelta.class);
+
+            transformer.transform(disqualifiedOfficersDelta);
+        } catch (RetryableErrorException ex) {
+            retryDeltaMessage(chsDelta);
+        } catch (Exception ex) {
```

Comment:
Should this be a NonRetryableErrorException? Otherwise what's the reason for having it?

Reply:
It's thrown as part of the api response in the insolvency consumer if the http response status is a 400. Technically the class for the exception doesn't have to be in this PR. Just thought since I was adding the retryable exception I'd do the leg work for the other exception to add in later. Happy to remove?